src/path_converters.h

```diff
@@ -789,22 +847,47 @@ class PathSimplifier : protected EmbeddedQueue<9>
     double m_origdx;
     double m_origdy;
     double m_origdNorm2;
-    double m_dnorm2Max;
-    bool m_lastMax;
+    double m_dnorm2ForwardMax;
+    double m_dnorm2BackwardMax;
+    bool m_lastForwardMax;
+    bool m_lastBackwardMax;
+    bool m_needToPushBack;
     double m_nextX;
     double m_nextY;
-    double m_lastWrittenX;
-    double m_lastWrittenY;
+    double m_nextBackwardX;
+    double m_nextBackwardY;
+    double m_currVecStartX;
+    double m_currVecStartY;
 
     inline void _push(double *x, double *y)
     {
-        queue_push(agg::path_cmd_line_to, m_nextX, m_nextY);
+        m_needToPushBack = (m_dnorm2BackwardMax > 0.0);
```

Comment:
Thanks for reviewing. This should be fixed in the new (forcefully rebased) commits.